addons/hr_timesheet/security/hr_timesheet_security.xml

```diff
@@ -97,10 +94,9 @@
             <field name="model_id" ref="model_timesheets_analysis_report"/>
             <field name="domain_force">[
                 ('user_id', '=', user.id),
-                '|', '|',
+                '|',
                     ('project_id.privacy_visibility', '!=', 'followers'),
-                    ('project_id.message_partner_ids', 'in', [user.partner_id.id]),
-                    ('task_id.message_partner_ids', 'in', [user.partner_id.id])
+                    ('message_partner_ids', 'in', [user.partner_id.id])
```

Comment:
With the current implementation, I think the timesheet user will no longer see something  if the privacy_visibility of the project is `followers`

Reply:
You also forgot to alter `timesheet_analysis_report_approver` rule. 🙂

Reply:
> With the current implementation, I think the timesheet user will no longer see something if the privacy_visibility of the project is followers

the search field is equivalent to `|, (task_id.message_partner_ids), (project_id.message_partner_ids)`, so this diff is equivalent. What is the concern her, am I missing something 🤔 

> You also forgot to alter timesheet_analysis_report_approver rule. 🙂

I didn't alter that rule, because it depends only on the followers of the project, but not on tasks. The search fields collects followers of both, so it wouldn't be an equivalency, I cannot replace it.

Reply:
For the report, since we don't have the compute/search method, the field will be always empty and so, the timesheet user will no longer see his records as he can for `account.analytic.line`. 🙂